### PR TITLE
Validate fragmented plan

### PR DESCRIPTION
**Pull request by arhimondr**
The first part of the https://github.com/prestodb/presto/pull/12419

### CODE REVIEW

**Comment by arhimondr** [comment]
The #12419 changes PlanFramenter so it is no longer trivial. It implements not trivial logic to expand ExchangeNode into multiple TableWriterNode and TableScanNode. It seems like a good idea to post validate the fragmented plan after introducing new non trivial plan nodes.

The PlanFragmented already introduces new nodes (RemoteSourceNode). But this plan modification is rather trivial.

File: presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java [hunk 1]
```diff
@@ -177,7 +178,7 @@ public void testTableScan()
     public void testProject()
     {
         TableScanNode tableScan = tableScan("ts", "orderkey");
-        PlanNode project = project("project", tableScan, "string", new Cast(new SymbolReference("orderkey"), "STRING"));
+        PlanNode project = project("project", tableScan, "string", new Cast(new SymbolReference("orderkey"), "VARCHAR"));
```
Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java]: unrelated change?

Comment (arhimondr) [reply presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java]: No, i had to fix that. The fragment created here was not passing the validation, as the type called `STRING` doesn't exist in Presto

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java [hunk 2]
```diff
@@ -42,6 +42,14 @@ public PlanSanityChecker(boolean forceSingleNode)
                         new NoSubqueryExpressionLeftChecker(),
                         new NoIdentifierLeftChecker(),
                         new VerifyOnlyOneOutputNode())
+                .putAll(
+                        Stage.FRAGMENT,
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java]: `FRAGMENTED`? Looks like it should be an adjective.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java [hunk 3]
```diff
@@ -66,13 +74,18 @@ public void validateIntermediatePlan(PlanNode planNode, Session session, Metadat
         checkers.get(Stage.INTERMEDIATE).forEach(checker -> checker.validate(planNode, session, metadata, sqlParser, types, warningCollector));
     }
 
+    public void validatePlanFragment(PlanNode planNode, Session session, Metadata metadata, SqlParser sqlParser, TypeProvider types, WarningCollector warningCollector)
```
[the remark this answers is not in the document]
Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java]: Final plan is the plan before fragmenter. One of the requiremenst for the final plan - is that one has to have an OutputNode, what is not true for a single PlanFragment.

[the remark this answers is not in the document]
Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java]: `validateFragmentedPlan` - i would like to keep it as `validatePlanFragment`, to emphasize that this method is inteded to validate only a single frament as opposed to the entire fragmented plan

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java (hunk 2 above)
Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanSanityChecker.java]: Ditto
